### PR TITLE
Server test. Combined image and directory extractors using shared storage

### PR DESCRIPTION
**Pull request by MashaSS**
### Motivation and context

Test scenario https://github.com/openvinotoolkit/cvat/issues/3123
Fixed in https://github.com/openvinotoolkit/cvat/pull/3424
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by bsekachev** [comment]
Please, merge develop. The PR shows extra changes.

**Comment by MashaSS** [comment]
> Please, merge develop. The PR shows extra changes.

Done.

File: cvat/apps/engine/tests/test_rest_api.py [hunk 1]
```diff
@@ -5293,6 +5293,84 @@ def test_api_v1_server_share_no_auth(self):
...
+            "server_files[5]": "/data1/street.png",
+            "server_files[6]": "/data1/subdir/",
+            "remote_files": [],
+            "use_zip_chunks": False,
+            "use_cache": False,
+            "copy_data": False
+        }
+
+        # create task with server
+        self._create_task(task, image_data)
```
Comment (bsekachev) [inline cvat/apps/engine/tests/test_rest_api.py:5372]: Can we also check a number of images in the created task?

File: cvat/apps/engine/tests/test_rest_api.py [hunk 2]
```diff
@@ -5293,6 +5293,84 @@ def test_api_v1_server_share_no_auth(self):
...
+    def setUpTestData(cls):
+        create_db_users(cls)
+
+    @staticmethod
+    def _create_shared_files():
+        main_path = os.path.join(settings.SHARE_ROOT, "data1")
+        os.makedirs(main_path)
+        main_images = ["people.jpeg", "street_1.jpeg", "street_2.jpeg", "street_3.jpeg"]
+        for img in main_images:
+            image = Image.new('RGB', size=(100, 50))
```
Comment (bsekachev) [inline cvat/apps/engine/tests/test_rest_api.py:5310]: I suppose we can create the image once before the loop

File: cvat/apps/engine/tests/test_rest_api.py [hunk 3]
```diff
@@ -5293,6 +5293,84 @@ def test_api_v1_server_share_no_auth(self):
...
+            image.save(osp.join(main_path, img), "JPEG")
+        image.save(osp.join(main_path, "street.png"), "PNG")
```
Comment (bsekachev) [inline cvat/apps/engine/tests/test_rest_api.py:5312]: Can we get file extension from image name (with os.path.splitext) and put the operator to the loop above?

File: cvat/apps/engine/tests/test_rest_api.py [hunk 4]
```diff
@@ -5293,6 +5293,84 @@ def test_api_v1_server_share_no_auth(self):
...
+        subdir_path = os.path.join(main_path, "subdir")
+        os.makedirs(subdir_path)
+        subdir_images = ["image_4.jpeg", "image_5.jpeg", "image_6.jpeg"]
+        for img in subdir_images:
+            image = Image.new('RGB', size=(100, 50))
```
Comment (bsekachev) [inline cvat/apps/engine/tests/test_rest_api.py:5317]: The same comment like above

**Comment by MashaSS** [comment]
@bsekachev all comments are resolved.

**Review by bsekachev — APPROVED** [review]

**Pull request merged**